src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -186,27 +203,46 @@ public void on(final PlayerInteractEvent event)
             if (block instanceof AbstractBlockHut
                     && !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
             {
-                cancelEvent(event);
+                cancelEvent(event, event.getEntityPlayer());
             }
 
-            if(Configurations.enableColonyProtection && (event.getWorld().getBlockState(event.getPos()).getBlock() instanceof BlockContainer
-                    || event.getWorld().getTileEntity(event.getPos()) != null || (event.getItemStack() != null && event.getItemStack().getItem() instanceof ItemPotion)))
+            final Permissions perms = colony.getPermissions();
+
+            if(isFreeToInteractWith(event.getWorld().getBlockState(event.getPos()).getBlock(), event.getPos())
```

Comment:
This method catches all PlayerInteractEvents. That means that it catches also the entity interact event. Handling this here is not a good idea.